ml-dhs-aws/src/main/java/com/marklogic/client/dhs/JobRunner.java

```diff
@@ -17,73 +17,92 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.sql.Timestamp;
 import java.time.LocalDateTime;
 import java.util.Iterator;
 import java.util.concurrent.atomic.AtomicLong;
 
 import org.slf4j.Logger;
 
-public class JobRunner {
-   private String jobId;
-   private String jobDirectory;
-   private int batchSize = 100;
-   private String[] roles;
-   private int timeIntervalInSeconds = 10;
-   private AtomicLong lastSuccessLogTime;
-   private AtomicLong lastFailureLogTime;
-   private Logger logger;
-   
-   public int getTimeIntervalInSeconds() {
-	return timeIntervalInSeconds;
-}
-public void setTimeIntervalInSeconds(int timeIntervalInSeconds) {
-	this.timeIntervalInSeconds = timeIntervalInSeconds;
-}
+public class JobRunner implements JobLogger{
```

Comment:
Should JobRunner be initialized with a JobLogger (similar to the current initialization with Logger) instead of implementing JobLogger?